### ISSUE TITLE
No support for alternative syntax of TypedDict

### ISSUE THREAD

**Issue by alukach** [microsoft/pyright#730]
# Issue

The Python3 [docs for the `TypedDict`](https://docs.python.org/3/library/typing.html#typing.TypedDict) states:

> To allow using this feature with older versions of Python that do not support PEP 526, TypedDict supports two additional equivalent syntactic forms:
>
> ```py
> Point2D = TypedDict('Point2D', x=int, y=int, label=str)
> Point2D = TypedDict('Point2D', {'x': int, 'y': int, 'label': str})
> ```

My expectation would be that all of the following examples would be supported.

## Example 1 ✅ 

```py
from typing import TypedDict

Point2D = TypedDict("Point2D", {"x": int, "y": int, "label": str})
```

This works as expected.

## Example 2 ❌

```py
from typing import TypedDict

Point2D = TypedDict("Point2D", x=int, y=int, label=str)
```

This does not work as expected:

```
  3:41 - error: Expected "total" parameter to have a value of True or False
  3:52 - error: Extra TypedDict arguments not supported
  3:11 - error: Expected dict as second parameter
```

## Example 3 ❌

```py
from typing import TypedDict

Point2D = TypedDict("Point2D", dict(x=int, y=int, label=str))
```

This does not work. In my mind, `TypedDict("Point2D", dict(x=int, y=int, label=str)) == TypedDict("Point2D", {"x": int, "y": int, "label": str})` but maybe there's something deeper involved in Python's typing system that makes this not true?

```
  3:11 - error: Expected dict as second parameter
```

# Context

I'm using `pyright 1.1.42`, experiencing the above issues both CLI and VSCode.

**Comment by erictraut** [comment]
Thanks for reporting the issue. The syntax specified in Example 1 is the one specified in PEP 526, which is the documentation I followed when I added support for TypedDict.

The syntax specified in Example 2 is not mentioned in PEP 526, but it appears that support for it was added in the final implementation. I don't understand why the implementers decided that it was useful to support a second equivalent syntax, but apparently there was some reason behind this decision. I can add this support.

Example 3 is not equivalent to example 1. By using a call to "dict" within the expression, details about the dict's type are erased, and it effectively becomes a Dict[str, Any].

I don't plan to add support for the variant in Example 3.

Consider a fourth example:
```
def function_that_returns_a_dict():
   return dict(x=int, y=int, label=str)
Point2D = TypedDict("Point2D", function_that_returns_a_dict())
```
This would run without generating an exception, but I think you can see why it's not reasonable to expect a static type checker to be able to be able to evaluate such expressions.

**Comment by erictraut** [comment]
This support is now included in Pyright 1.1.43, which I just published.
